src/check_release.rs

```diff
@@ -374,6 +375,7 @@ pub(super) fn run_check_release(
                 true,
             )?;
         } else {
+            writeln!(config.stderr())?;
```

Comment:
This blank line appears to be printed if there were no errors, whether or not any warnings were produced. If no warnings were printed, I'm not sure we want a blank line there.

To me, conceptually, the blank line is the last line of each warning or error block, not a prefix to the summary line. So if there were no error or warning blocks (`--- failure ...`) then I probably wouldn't include a blank line here.

Reply:
This is inside `check_release.rs:299`: `if produced_errors || produced_warnings`, so it must be true that if no errors are produced, some warnings are produced, although this isn't expressed well in the code.  Would it make sense to change the `else` to an `else if warnings_produced {...} else { unreachable!() }` to be more expressive?

Reply:
Yes, please! That would definitely have been more clear, and it's safer against future refactoring too.